src/RootScene.js

```diff
@@ -72,6 +74,17 @@ export default class RootScene extends Component {
     return route.transition || this.props.defaultTransition;
   }
 
+  handleRouteChange = () => {
+    const navigator = this.getNavigator();
+    // Stack diff can be either 0 or 2, but never 1.
+    if (
+      (navigator && this.routeStack && navigator.state.routeStack.length) ===
```

Comment:
why these parenthesis?
we should check that the navigator exists, then check if we have tracked the routeStack. If those are ok, then check if the stacks length are ok.
This should be
`if (navigator && this.routeStack && navigator.state.routeStack.length === this.routeStack.length + 1)`